src/relax/op/tensor/binary.h

```diff
@@ -48,13 +48,15 @@ namespace relax {
       .add_argument("x1", "Tensor", "The first input tensor.")     \
       .add_argument("x2", "Tensor", "The second input tensor.")
 
-#define RELAX_REGISTER_BINARY_BROADCAST_OP_AND_IMPL(OpName)             \
-  RELAX_REGISTER_BINARY_OP_AND_IMPL(OpName).set_attr<FInferStructInfo>( \
-      "FInferStructInfo", InferStructInfoBroadcastArith)
-
-#define RELAX_REGISTER_CMP_OP_AND_IMPL(OpName)                          \
-  RELAX_REGISTER_BINARY_OP_AND_IMPL(OpName).set_attr<FInferStructInfo>( \
-      "FInferStructInfo", InferStructInfoBroadcastCMP)
+#define RELAX_REGISTER_BINARY_BROADCAST_OP_AND_IMPL(OpName)                          \
+  RELAX_REGISTER_BINARY_OP_AND_IMPL(OpName)                                          \
+      .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoBroadcastArith) \
+      .set_attr<FRelaxInferLayout>("FRelaxInferLayout", InferLayoutBinaryEwise)
+
+#define RELAX_REGISTER_CMP_OP_AND_IMPL(OpName)                                     \
+  RELAX_REGISTER_BINARY_OP_AND_IMPL(OpName)                                        \
+      .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoBroadcastCMP) \
+      .set_attr<FRelaxInferLayout>("FRelaxInferLayout", InferLayoutBinaryEwise)
```

Comment:
I see we have the same layout inference func for them. We can put it to the place just below Line 49.